src/neo4j_graphrag/experimental/components/pdf_loader.py

```diff
@@ -85,16 +82,21 @@ def load_file(
 
     async def run(
         self,
-        filepath: Path,
+        filepath: Union[str, Path],
         metadata: Optional[Dict[str, str]] = None,
-        fs: Optional[AbstractFileSystem] = None,
+        fs: Optional[Union[AbstractFileSystem, str]] = None,
     ) -> PdfDocument:
-        fs = fs or LocalFileSystem()
+        if not isinstance(filepath, str):
+            filepath = str(filepath)
+        if isinstance(fs, str):
```

Comment:
Why pass a str for the filesystem and initialise it here and not just pass the `fsspec.filesystem` to the method directly? We could remove the `Union` part of the type hint if we did. We could also set the default to `LocalFileSystem`

Reply:
The reason for the string is to make it work with config files. We could have a parser in the config parser to initialize a file system though, but I thought it was easier to just support str since this is allowed by `fsspec`.

Reply:
Makes sense. I don't have a strong view on whether or not it should be here or in the config parser so let's just keep it here for now